### PR TITLE
Minor bugfix in calculation complex transport demand.

### PR DESCRIPTION
**Pull request by MariannaR**
Missing time steps are added to the extrapolation of the demand for `complex` based on EDGE-T values.

### CODE REVIEW

File: R/readEDGETransport.R [hunk 1]
```diff
@@ -307,7 +307,7 @@ readEDGETransport <- function(subtype = "logit_exponent") {
            tmp[node == "Electric Trains", node := "ueelTt"]
            ## extend to time steps necessary for the input demand trend
            tmp = approx_dt(tmp,
-                           xdata = c(seq(1993, 2010, 1), seq(2105, 2150, 5)),
+                           xdata = c(seq(1993, 2010, 1), unique(tmp$year)[unique(tmp$year)>2010], seq(2105, 2150, 5)),
```
Comment (Loisel) [inline R/readEDGETransport.R:310]: why not just `seq(2015, 2150, 5)`?

Comment (Loisel) [reply R/readEDGETransport.R:310]:
```suggestion
                           xdata = c(seq(1993, 2010, 1), seq(2015, 2150, 5)),
```

**Review by Loisel — CHANGES_REQUESTED** [review]
In principle this is fine but I want to test the code change suggestions :)

File: R/readEDGETransport.R (hunk 1 above)
Comment (MariannaR) [reply R/readEDGETransport.R:310]: because REMIND does not require 5 years timesteps after 2050

**Review by Loisel — APPROVED** [review]

File: R/readEDGETransport.R [hunk 2]
```diff
@@ -294,11 +294,10 @@ readEDGETransport <- function(subtype = "logit_exponent") {
...
            tmp = tmp[grepl("ConvCase", EDGE_scenario)]
            ## convert from final energy to useful energy
-           tmp[fuel == "BEV", totdem := totdem*0.64] ## battery electric LDV
-           tmp[fuel == "FCEV", totdem := totdem*0.36] ## FCEV vehicles LDV and HDV
-           tmp[grepl("Liquids|NG", fuel) & node == "LDV", totdem := totdem*0.22] ## ICE LDV
-           tmp[grepl("Liquids|NG", fuel) & node == "HDV", totdem := totdem*0.24] ## ICE HDV
-           tmp[grepl("Electric", fuel) & node == "HDV", totdem := totdem*0.64] ## battery electric HDV
+           tmp[fuel == "BEV", totdem := totdem*3] ## battery electric LDV
+           tmp[fuel == "FCEV" & node == "LDV", totdem := totdem*2.5] ## FCEV vehicles LDV
+           tmp[fuel == "FCEV" & node == "HDV", totdem := totdem*1.5] ## FCEV vehicles HDV
```
Comment (Loisel) [inline R/readEDGETransport.R:299]: Where does this factor for FCEV HDVs come from?

Comment (MariannaR) [reply R/readEDGETransport.R:299]: Robert suggested both the factors for FCEV and electric HDVs. They should take into account the fact that in `complex` HDVs are and average of all transport heavy duty categories and of different efficiencies.

**Review by Loisel — APPROVED** [review]

**Pull request merged**